Bevy 0.15

csandven opened the pull request.

csandven commented:
@malbernaz  Ant chance for a review and potential crate publishing?

malbernaz commented:
> @malbernaz Any chance for a review and potential crate publishing?

Absolutely! I'll check it out. Thanks for this :)

malbernaz commented on src/extract.rs:
```diff
@@ -50,31 +50,22 @@ pub struct ExtractedLightOccluder2d {
+        Query<(RenderEntity, &LightOccluder2d, &GlobalTransform, &ViewVisibility)>,
     >,
 ) {
     commands.spawn(ExtractedLightOccluder2d::default());
```
This is adding an empty `ExtractedLightOccluder2d` that is not needed any more. In fact if you run the example you'll find an artifact in the middle of the screen that is a ghost occluder. The loop in `sdf.wgsl:27` as it is accounts for this empty element and starts at element `2u` (I cannot recall why 2). It should start from `0u` from now on.

My bad that this is poorly commented. It's related to how GPU buffers are created in bevy.

malbernaz commented on src/extract.rs:
```diff
@@ -88,30 +79,21 @@ pub struct ExtractedPointLight2d {
-    point_lights_query: Extract<Query<(Entity, &PointLight2d, &GlobalTransform, &ViewVisibility)>>,
+    point_lights_query: Extract<Query<(RenderEntity, &PointLight2d, &GlobalTransform, &ViewVisibility)>>,
 ) {
     commands.spawn(ExtractedPointLight2d::default());
```
This is also adding an empty entity that should also be removed from `lighting.wgsl:42`.

malbernaz commented on Cargo.toml:
```diff
@@ -11,15 +11,17 @@ readme = "README.md"
   "bevy_winit",
   "x11",
 ] }
+bevy_image = "0.15.0"
```
We can solve it by adding `png` to the bevy features. Can you do this and add a todo comment saying that we should revisit it after the bevy issue is solved.

malbernaz commented on src/types.rs:
```diff
@@ -95,24 +101,20 @@ impl Default for PointLight2d {
 }
 
 /// A bundle of components representing a point light in a 2D environment.
+#[deprecated]
```
```rust
#[deprecated(
    since = "0.4.0",
    note = "Use the `PointLight2d` component instead."
)]
```

malbernaz commented on src/types.rs:
```diff
@@ -126,17 +128,12 @@ impl LightOccluder2d {
 
 /// A bundle of components representing a light occluder in a 2D environment.
 #[derive(Bundle, Default)]
+#[deprecated]
```
```rust
#[deprecated(
    since = "0.4.0",
    note = "Use the `LightOccluder2d` component instead."
)]
```

malbernaz commented on examples/basic.rs:
```diff
@@ -43,22 +44,19 @@ fn setup(mut commands: Commands) {
-                ..default()
+            Sprite {
+                custom_size: Some(Vec2::splat(100.0)),
+                ..Default::default()
```
`..default()` instead for consistency.

malbernaz commented on examples/basic.rs:
```diff
@@ -18,7 +19,7 @@ struct MovingLights;
 fn setup(mut commands: Commands) {
     commands.spawn((
-        Camera2dBundle::default(),
+        Camera2d::default(),
```
We can drop the `::default()` here.

malbernaz reviewed (commented):
Overall this is looking really good, thanks a lot!

I need to test this more thoroughly for `webgpu`  and `webgl` targets. Once it's done we can merge and publish it.

Also can you run `cargo fmt` on the files that were modified?

csandven commented:
Thanks for the comments. Added fixes now. Also added a rustfmt.toml as the one I have for work i really different than what is used normally. Hope it is fine!

malbernaz reviewed (changes requested):
The only two changes I requested are to make everything work correctly on web targets.

The context is that `GpuArrayBuffer` is probably initialising the array buffers with an empty value - I have to check bevy's source to understand exactly what has changed since `0.14`. This behaviour creates an empty point light and an empty occluder that was visible on the screen.

Thanks a lot for the contribution! After merged I'll increment the changelog and publish the update. 👏👏👏

malbernaz commented:
Released! Thanks ❤️
